### DISCUSSION TITLE
Add option to copy Full Path with \ before spacings

### DISCUSSION THREAD

**Discussion by timur-g** [doublecmd/doublecmd#1130] [Suggestions]
### Description

DC has copy of Full Path, Ctrl+shift+C, but in Linux it may be something like "/abc/file with spacing.ext".
In order to use it in the console, it should look like "/abc/file\ with\ spacing.ext".
Can you please add an option that copy of Full Path copies ready for console?

**Comment by Skif-off** [comment]
What about [Lua](https://doublecmd.github.io/doc/en/lua.html)? Something like

```lua
-- (cross-platform)
--
--[[
Copy to the clipboard the full name of the file under the cursor and escape spaces:
quotes in Windows or "\" in other cases.

How to use:
internal command: cm_ExecuteScript
params:
  path-to-script
  %"0%p0
]]

local params = {...}

if #params ~= 1 then
  local sScrName = debug.getinfo(1).source
  if string.sub(sScrName, 1, 1) == '@' then
    sScrName = string.sub(sScrName, 2, -1)
  end
  sScrName = SysUtils.ExtractFileName(sScrName)
  Dialogs.MessageBox("Check the number of parameters!", sScrName, 0x0030)
  return
end

if string.find(params[1], " ", 1, true) ~= nil then
  if SysUtils.PathDelim == "\\" then
    Clipbrd.SetAsText('"' .. params[1] .. '"')
  else
    local sTmp = string.gsub(params[1], " ", "\\ ")
    Clipbrd.SetAsText(sTmp)
  end
else
  Clipbrd.SetAsText(params[1])
end
```
Save as, for example, `test.lua ` and add button (see details in the beginning of script).

**Comment by timur-g** [comment]
@Skif-off Thanks for the script, will be handy and help me use this. 
Just curious, Hotkeys have parameters, was it possible to add some to Ctrl+Shift+C to get the same result?

I would still prefer an option, as more simple approach - no pressing now that this workaround is here.

**Comment by Skif-off** [comment]
> Just curious, Hotkeys have parameters, was it possible to add some to Ctrl+Shift+C to get the same result?

Sorry, I'm not sure I understood the question :)
Currently [cm_CopyNamesToClip](https://doublecmd.github.io/doc/en/cmds.html#cm_CopyNamesToClip) / [cm_CopyFullNamesToClip](https://doublecmd.github.io/doc/en/cmds.html#cm_CopyFullNamesToClip) have no parameters.
For this or a similar script, you can add a hotkey (Ctrl+Alt+C?) or a toolbar button and a hotkey. (Use `cm_ExecuteScript` in both cases.)

**Comment by timur-g** [comment]
Sorry for not being clear. Your script is working (and I figuread out this is the best as I can have both spacings and \), just wondering for other approach.
Hotkeys *can* have parameters. Was it possible to modify existing Ctrl+Shift+C hotkey with some parameter to achieve the same result?

**Comment by Skif-off** [comment]
> Hotkeys can have parameters.

No, internal [commands](https://doublecmd.github.io/doc/en/cmds.html) have parameters.
We can
...set one or more hotkeys for one command,
...set several hotkeys for one command, but specifying different parameters for this command in each case.
